core/templates/dev/head/domain/classifier/AnswerClassificationResultObjectFactorySpec.ts

```diff
@@ -16,25 +16,67 @@
  * @fileoverview Unit tests for the AnswerClassificationResultObjectFactory.
  */
 
-require('domain/classifier/AnswerClassificationResultObjectFactory.ts');
+import { AnswerClassificationResultObjectFactory } from
+  'domain/classifier/AnswerClassificationResultObjectFactory.ts';
+
 require('domain/exploration/OutcomeObjectFactory.ts');
 require(
   'pages/exploration-player-page/services/answer-classification.service.ts');
 
-describe('Answer classification result object factory', function() {
-  var oof, acrof;
-  var DEFAULT_OUTCOME_CLASSIFICATION;
+class MockSubtitledHtml {
+  _html: any;
+  _contentId: any;
+  constructor(html, contentId) {
+    this._html = html;
+    this._contentId = contentId;
+  }
+}
+
+class MockOutcome {
+  dest: any;
+  feedback: any;
+  labelledAsCorrect: any;
+  paramChanges: any;
+  refresherExplorationId: any;
+  missingPrerequisiteSkillId: any;
+  constructor(
+      dest, feedback, labelledAsCorrect, paramChanges,
+      refresherExplorationId, missingPrerequisiteSkillId) {
+    this.dest = dest;
+    this.feedback = feedback;
+    this.labelledAsCorrect = labelledAsCorrect;
+    this.paramChanges = paramChanges;
+    this.refresherExplorationId = refresherExplorationId;
+    this.missingPrerequisiteSkillId = missingPrerequisiteSkillId;
+  }
+}
 
-  beforeEach(angular.mock.module('oppia'));
+class MockOutcomeObjectFactory {
+  createNew(
+      dest = null, feedbackTextId = null, feedbackText = null,
```

Comment:
Why does this need default args? It would probably be better not to have those, since outcomes with feedbackText = null (say) aren't actually real things.

Reply:
This was done because if you see [here](https://github.com/oppia/oppia/pull/7145/files/6fa6c50d7f32fdbe5ee0885ef2be25f5d8950057#diff-3c12e29f37fcd1dc60a5ff9f2c2e49f8R85) `MockOutcomeObjectFactory` is instantiated with just three args whereas it needs four, this does not go down well with the TS class and it starts complaining. That's why default args fill in when situations like these arise.

Reply:
No, don't use default args. Fix the arguments so that the correct number of arguments are passed.

Default args lead to additional (hidden) branching and more errors. Harder to maintain.

Reply:
Done!